src/Compilers/VisualBasic/Test/Symbol/DocumentationComments/DocCommentTests.vb

```diff
@@ -12115,10 +12115,10 @@ xmlDoc)
                        DocumentationMode.Diagnose,
                        DocumentationMode.Parse))
 
-            Dim compilation = CreateCompilation(sources,
-                                                additionalRefs,
-                                                TestOptions.ReleaseDll.WithXmlReferenceResolver(XmlFileResolver.Default),
-                                                parseOptions)
+            Dim compilation = CreateCompilationWithMscorlib40(sources,
```

Comment:
>CreateCompilationWithMscorlib40 [](http://example.com/codeflow?start=30&length=31)

The reason behind this change is not obvious

Reply:
Made this change more granular here. But if you are asking about the NoPIA test changes in general - I think the tests were wrong because they were not referencing mscorlib40 everywhere which caused the `ImportedFromTypeLibAttribute` to be missing and hence the `AttributeConstructor` was `null` when decoding the attribute.

Reply:
> This "wrong PIA usage" worked with the previous metadata cracking method, so not sure if it's fine for it to not work anymore.

Could you provide more details about how exactly the original test was affected? Otherwise it is hard to judge whether the behavior change is acceptable.

Reply:
Let's take test `EmitMetadata.InParametersShouldHaveMetadataIn_NoPIA` as an example. The `comAssembly` compilation references `Net40.mscorlib`. The second compilation references `NetStandard20.mscorlib` on `net8.0` and `Net461.mscorlib` on `net472`. The problem happens on `net8.0` where the mscorlib doesn't seem to have the type `ImportedFromTypeLibAttribute` defined. The corresponding `PEAttributeData` have `AttributeClass` pointing to an `ErrorType` which says "error CS7069: Reference to type 'ImportedFromTypeLibAttribute' claims it is defined in 'mscorlib', but it could not be found" and hence `AttributeConstructor` is `null`, so the `ImportedFromTypeLibAttribute` cannot be decoded and the whole compilation then fails with "error CS1759: Cannot embed interop types from assembly '..., Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' because it is missing either the 'System.Runtime.InteropServices.ImportedFromTypeLibAttribute' attribute or the 'System.Runtime.InteropServices.PrimaryInteropAssemblyAttribute' attribute.". Making the second compilation reference the correct mscorlib fixes this problem.

Reply:
I think we should stop relying on `GetAttributes` to check the presence of well-known attributes in `ValidateLinkedAssemblies`. The pattern that we usually follow is to have a dedicated API on symbol (`AssemblySymbol` in this case), which gets data from source for source assembly, and gets data from metadata (`PEModule`, which doesn't need symbols for well-known attributes) for PE assemblies. The scenarios should work without any changes to the tests.